janitor/main.go

```diff
@@ -145,27 +150,41 @@ func run() error {
 		"terminateNode.timeout", cfg.TerminateNode.Timeout,
 		"global.manualMode", cfg.Global.ManualMode)
 
-	// Start a simple HTTP server for the /config endpoint
-	go func() {
-		mux := http.NewServeMux()
-		mux.HandleFunc("/config", func(w http.ResponseWriter, r *http.Request) {
-			w.Header().Set("Content-Type", "application/json")
+	// Parse config port from address
+	// Handles formats like ":8082", "localhost:8082", "0.0.0.0:8082"
+	_, portStr, err := net.SplitHostPort(configAddr)
+	if err != nil {
+		// If SplitHostPort fails, assume it's just a port number
+		portStr = configAddr
+		if portStr != "" && portStr[0] == ':' {
+			portStr = portStr[1:]
+		}
+	}
 
-			if err := json.NewEncoder(w).Encode(cfg); err != nil {
-				slog.Error("Failed to encode configuration as JSON", "error", err)
-				http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+	configPort, err := strconv.Atoi(portStr)
+	if err != nil {
+		slog.Error("Invalid config-bind-address port", "error", err, "address", configAddr)
+		return fmt.Errorf("invalid config-bind-address port %q: %w", configAddr, err)
+	}
 
-				return
-			}
-		})
+	// Create config handler
+	configHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
 
-		slog.Info("Starting config HTTP server", "address", configAddr)
+		if err := json.NewEncoder(w).Encode(cfg); err != nil {
+			slog.Error("Failed to encode configuration as JSON", "error", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 
-		// nolint:gosec // G114: Config endpoint is for internal debugging/monitoring
-		if err := http.ListenAndServe(configAddr, mux); err != nil {
-			slog.Error("Config HTTP server failed", "error", err)
+			return
 		}
-	}()
+	})
+
+	// Create config server using common server implementation
+	configServer := server.NewServer(
+		server.WithPort(configPort),
+		server.WithHandler("/config", configHandler),
+		server.WithSimpleHealth(),
```

Comment:
I think the controller already has some health checks -- 

```
// Setup health checks
	if err = mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
		slog.Error("Unable to set up health check", "error", err)
		return err
	}
	if err = mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
		slog.Error("Unable to set up ready check", "error", err)
		return err
	}
```

can we integrate the same here instead of having a different health check?